orsopy/fileio/base.py

```diff
@@ -192,7 +194,7 @@ def _resolve_type(hint: type, item: Any) -> Any:
                 except ValueError:
                     # string wasn't ISO8601 format
                     return None
-            if issubclass(hint, Header):
+            if issubclass(hint, Header) and hasattr(item, "keys"):
```

Comment:
Why not `isinstance(item, dict)`, which seems clearer to me?

Reply:
I was not sure if we can assume that the item is actually a dict and not a dict like object. This seemed to me a bit more general but I checked now and all relevant other classes like OrderedDict are actually subclasses of dict and this should work.